exec/java-exec/src/test/java/org/apache/drill/TestPartitionFilter.java

```diff
@@ -376,4 +376,14 @@ public void testPartitionFilterWithLike() throws Exception {
     testIncludeFilter(query4, 4, "Filter", 16);
   }
 
+  @Test //DRILL-3710 Partition pruning should occur with varying IN-LIST size
+  public void testPartitionFilterWithInSubquery() throws Exception {
+    String query = String.format("select * from dfs_test.`%s/multilevel/parquet` where cast (dir0 as int) IN (1994, 1994, 1994, 1994, 1994, 1994)", TEST_RES_PATH);
+    /* In list size exceeds threshold - no partition pruning since predicate converted to join */
+    test("alter session set `planner.in_subquery_threshold` = 2");
```

Comment:
A bug could cause us to not obey the option at all i.e. we always do partition pruning regardless of the option setting. This unit test checks we do obey the option.